MH-12615 Improve the languages drop-down menu

KatrinIhler opened the pull request:
A default languages can now be configured in languages.properties (this possibility extends to other listproviders if they implement the necessary methods).
The drop-down menus for all metadata fields are now sorted.

Commit:
MH-12615 Add the possibility of a default value to the listproviders

Commit:
MH-12615 Get key of default value from listproviders if defined

Commit:
MH-12615 Write default in metadata field

Commit:
MH-12615 Delete dead code

Commit:
MH-12615 Sort metadata drop-down menus

Commit:
MH-12615 Move hardcoded listproviders to files

karendolan commented on modules/index-service/src/main/java/org/opencastproject/index/service/resources/list/api/ResourceListProvider.java:
```diff
@@ -56,5 +56,14 @@
+  /**
+   * Returns the key of the default value in the list if customized, else null.
+   *
+   * @return key of default value or null //TODO Or is blank better?
```
Making this method concrete hides the decision from the Impl programmer. IMHO, it would be better to leave this method as an API which requires the Implementation programmer to consider what makes the most sense for the type of Implementation that implements this API. 

The program that calls this method cannot make assumptions if this will be blank or null because the Impl programer can always override the method to return a blank or null.

KatrinIhler replied:
I agree with the second part. Regarding the first part of your comment: I decided to provide a default (hehe) implementation for getDefault() since a lot of the other list providers currently have no need for a default and I didn't want to bloat the code with a lot of "...return null". This is debatable of course.

staubesv commented on etc/listproviders/locations.properties:
```diff
@@ -0,0 +1,6 @@
+list.name=locationFilter
```
Thanks for factoring out the hard-coded lists, but as for this list, I don't think that it is of any use. Please remove the list "locationFilter" or provide a reason why you think it is needed.

staubesv commented on etc/listproviders/event.sources.properties:
```diff
@@ -0,0 +1,5 @@
+list.name=eventSourcesFilter
+list.translatable=false
```
Thanks for factoring out the hard-coded lists, but as for this list, I don't think that it is of any use. Please remove the list "eventSourcesFilter" or provide a reason why you think it is needed.

KatrinIhler replied:
You're right, these lists are no longer in use, I overlooked that. Will remove them.

staubesv commented on etc/listproviders/event.status.properties:
```diff
@@ -0,0 +1,9 @@
+list.name=eventStatusFilter
```
Thanks for factoring out the hard-coded lists, but as for this list, I don't think that it is of any use. Please remove the list "eventStatusFilter" or provide a reason why you think it is needed.

staubesv commented on modules/index-service/src/main/java/org/opencastproject/index/service/resources/list/api/ResourceListProvider.java:
```diff
@@ -56,5 +56,14 @@
+   *
+   * @return key of default value or null
+   */
+  public String getDefault() {
```
As follow-up to the previous comments of @karendolan:

Personally, I share the preference of @karendolan to don't hide this from the implementation, but there are arguments pro and contra that.

What should not be the case, however, is that different functions of the smae interface handle this decision differently: The same argument would hold for the method isTranslatable(), but in that case, each implementation does implement the method (even if a default would be "less lines" in overall).

Maybe quickly talk to @lkiesow (IIRC, he added isTranslatable) about his opinion.

lkiesow replied:
@wsmirnow  wrote that. I have no strong opinion on this btw.

KatrinIhler replied:
@staubesv I do see your point regarding the inconsistency between getDefault() and getTranslatable(). Since the majority seems to favor it, I will change getDefault() so that it's implemented in each provider.

staubesv reviewed (changes requested):
I've done with a first review round. The two main change requests are:

- Remove obsolete list providers from codebase instead of factoring them out
- Inconsistent "interface style"

Commit:
MH-12615 Remove unused list providers

Remove the unused list providers eventSourcesFilter, eventStatusFilter
and locationFilter. Additionally remove the related ConstantsService
which was tested, but never used.

Commit:
MH-12615 Implement getDefault in all listproviders

Make ResourceListProvider an interface instead of an abstract class and
implement getDefault() in all subclasses to be consistent with the
Implementation of isTranslatable().

staubesv reviewed (approved):
During testing I've found a number of issues related to this PR and created JIRA tickets for them:

- [MH-12669](https://opencast.jira.com/browse/MH-12669)
Events->"Event Details"->Metadata: Language not displayed in human-readable format in read-only mode
- [MH-12670](https://opencast.jira.com/browse/MH-12670) "Add Event"->Summary: Selected language not listed
- [MH-12671](https://opencast.jira.com/browse/MH-12671) It is not possible to unselect a language
- [MH-12672](https://opencast.jira.com/browse/MH-12672) Remove obsolete Admin UI component endpoint

None of these issues is caused by this PR (verified) and while related, the issues are not in scope to what this PR does.

-> Time for merging this PR!
